opacus/tests/grad_samples/conv2d_test.py

```diff
@@ -69,21 +70,29 @@ def test_conv2d(
             dilation=dilation,
             groups=groups,
         )
-        self.run_test(x, conv, batch_first=True, atol=10e-5, rtol=10e-4)
+        is_ew_compatible = padding != 'same'  # TODO add support for padding = 'same' with EW
```

Comment:
does "valid" padding work fine with EW?

Reply:
Good question. It should since it's basically zero padding. But let me run a test locally and verify this. Good catch!

Reply:
well, actually on line 37 of this file we have unit tests for both "valid" and "same". So it should work just fine 😎